### ISSUE TITLE
about the training

### ISSUE THREAD

**Issue by shawnthu** [WangYueFt/rfs#1]
In training phrase:
    1. first, merge a big and full data-set
    2. second, train a ordinary res-net classification model
    3. finally, discard the soft-max layer, keep the embedding
**that's all the training ?!**

To test the few-shot problem, suppose given 20 classes, and 10 samples for each class, 
construct a test N-way K-shot meta-task as follows:
    1. random pick N=5 classes, and random pick K=5 samples from each corresponding class as the **train set**, 25 in total
    2. random pick, e.g. 3 sample, from each class, to construct a **test set** with 15 samples within
    3. now we have a meta-task, then train linear classification with feature extracted from train set, using the embedding model above, then calculate the error on the test set.
Repeat the above 3 steps M times, merge and get a final test error.
**that's all the test ?!**

**Comment by valencebond** [comment]
i am also confused about the training stage and test stage. And i am not sure that “merge a big and full data-set“ is allowed in few-shot settings.

**Comment by HobbitLong** [comment]
Hi, @shawnthu  and @valencebond, I think that is a good question, and it actually reflects the gap between a realistic few-shot learning setting and current standard benchmark, as I think.

Nowadays, I see many meta-learning algorithms do two-step training: (1) pre-training by merging the full dataset with standard cross-entropy; (2) do meta-training. We show (1) is sufficient to achieve SOTA results with minimal whistle and bells. On top of this baseline, one may run a meta-learning algorithm and check its relative improvement, i.e., the effectiveness of step (2).

Another related question is: whether these widely used benchmark are realistic enough? IMO, classes are hierarchically structured in real-world, so we can not simply merge them to form a classification task. However, currently widely-used few-shot learning benchmarks, e.g., _miniImageNet_ and _CIFAR-FS_, have the prior of mutually exclusion between classes, which the pre-training stage can take advantage of. It seems leveraging this prior is enough to generate good embeddings for new task. So we also hope this baseline could spur re-examination of current benchmarks.

Hope this answer helps.

**Comment by valencebond** [comment]
hi @HobbitLong , thanks for your detailed explanation. i agree that the existing few-shot setting is not realistic. But my point is that unrealistic setting is caused by the training pipeline of constructing many small meta task consisting of N-way K-shot. i cannot map this pipeline into any real-world industry application. please correct me if there is something wrong.

**Comment by WangYueFt** [comment]
> hi @HobbitLong , thanks for your detailed explanation. i agree that the existing few-shot setting is not realistic. But my point is that unrealistic setting is caused by the training pipeline of constructing many small meta task consisting of N-way K-shot. i cannot map this pipeline into any real-world industry application. please correct me if there is something wrong.

Hi,

I think once the training data is given, any data split should be allowed to align training and testing. We do need to rethink how to design a better evaluation metric for few-shot learning, however, the current pre-training approach in few-shot learning doesn't sound fake to me. I'll close this issue and feel free to re-open if you have further questions.

Best,
Yue
